### PR TITLE
c/snap,asserts: start supporting ExternalKeypairManager in the snap key-related commands

### PR DESCRIPTION
**Pull request by pedronis**
This is done by centralizing keypair manager creation and a common KeypairManager interface.

### CODE REVIEW

**Comment by pedronis** [comment]
based on #10370

File: cmd/snap/complete.go [hunk 1]
```diff
@@ -185,13 +184,18 @@ func (n assertTypeName) Complete(match string) []flags.Completion {
...
+	if err != nil {
 		return nil
-	})
+	}
+	keys, err := keypairManager.List()
+	if err != nil {
+		return nil
+	}
+	res := make([]flags.Completion, len(keys))
+	for i, k := range keys {
```
Comment (mvo5) [inline cmd/snap/complete.go:196]: The old NewGPGKeypairManager based code has a `strings.HasPrefix(uid, match)` check here that the new code does not have anymore. Is this no longer needed?

File: cmd/snap/keymgr_test.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+var _ = check.Suite(&keymgrSuite{})
+
+func (keymgrSuite) TestGPGKeypairManager(c *check.C) {
+	keypairMgr, err := snap.GetKeypairManager()
+	c.Check(err, check.IsNil)
+	c.Check(keypairMgr, check.FitsTypeOf, &asserts.GPGKeypairManager{})
+}
+
+func (keymgrSuite) TestExternalKeypairManager(c *check.C) {
```
Comment (mvo5) [inline cmd/snap/keymgr_test.go:42]: (nitpick) maybe a test for the failure case here? i.e. if the external keymgr returns "exit 1" or something to see that the error looks sensible and the code is DTRT?

**Review by mvo5 — COMMENTED** [review]
Looks great, just one nitpick and one question about the completion handling.

**Comment by pedronis** [comment]
@mvo5 updated

**Review by anonymouse64 — APPROVED** [review]
I didn't review the previous PR for this, but this one LGTM on it's own merits

**Review by mvo5 — APPROVED** [review]
Thank you

**Pull request merged**